### DISCUSSION TITLE
Working with huge Hetero Data

### DISCUSSION THREAD

**Discussion by mprithiv** [pyg-team/pytorch_geometric#3972] [Q&A]
Hi

I currently have a huge hetero data am trying to build a HeteroData object using [this link](https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html). The number of nodes and features are huge that the object creation takes for ever. I am using `SentenceTransformer` for ahpha-numeric features, `LabelEncoder` for numeric and date features. Is there any way to build the graph more efficiently.

**Comment by rusty1s** [comment]
Can you say with part of the pipeline takes the most time? I think this mostly might be due to the usage of `SentenceTransformer` (whose efficiency is out of our scope).
